lib/src/router/coordinator.dart

```diff
@@ -1,19 +1,52 @@
 import 'package:flutter/material.dart';
 import 'package:get_it/get_it.dart';
-import 'package:myapp/src/router/auto_router.dart';
-import 'package:myapp/src/router/auto_router.gr.dart';
+import 'package:go_router/go_router.dart';
+import 'package:myapp/src/router/route_name.dart';
+import 'package:myapp/src/router/router.dart';
 
 class XCoordinator {
   static XRouter get rootRouter => GetIt.I<XRouter>();
+  static final navigatorKey = GlobalKey<NavigatorState>();
+  static final shellKey = GlobalKey<NavigatorState>();
```

Comment:
what is different between `shellKey` and `navigatorKey`

Reply:
`shellKey` maintains the state in `ShellRoute` (nested route), `navigatorKey` like the root key. With those keys, you can specify which screen can have bottom bar, etc.

```dart
final navigatorKey = GlobalKey<NavigatorState>();
final shellKey = GlobalKey<NavigatorState>();

|_ ShellRoute
  |_ GoRoute                            // Needs Bottom Navigation                  
    |_ parentNavigatorKey = shellKey   
  |_ GoRoute                            // Needs Bottom Navigation
    |_ parentNavigatorKey = shellKey   
|_ GoRoute                              // Full Screen which doesn't need Bottom Navigation
  |_parentNavigatorKey = navigatorKey
  ```